### PR TITLE
allow fuseki config per service (query, update and graphStore) and ad…

### LINKED ISSUE #26: Extend fuseki configuration per service and add Authentication
At the moment fuseki configuration just takes the URL of the endpoint. Add option to extend this with 1 URL per service (query, update and graphStore).
Also add the option to use a password protected fuseki service.

### PR DESCRIPTION
**Pull request by domivds**
…d auth support

### CODE REVIEW

File: cogni-libs/src/main/java/zone/cogni/libs/sparqlservice/impl/FusekiSparqlService.java [hunk 1]
```diff
@@ -17,42 +27,65 @@
 import java.io.IOException;
 import java.io.StringWriter;
 import java.nio.charset.StandardCharsets;
+import java.util.Base64;
 import java.util.Collections;
 import java.util.function.Function;
 
 public class FusekiSparqlService implements SparqlService {
```
Comment (natan-cox-cognizone) [inline cogni-libs/src/main/java/zone/cogni/libs/sparqlservice/impl/FusekiSparqlService.java:34]: why not use `ConfigurationProperties` so configuration is part of the component?

we could get rid of FusekiConfig this way.

Comment (natan-cox-cognizone) [reply cogni-libs/src/main/java/zone/cogni/libs/sparqlservice/impl/FusekiSparqlService.java:34]: I'm not even sure where and how `SparqlServiceProvider` get executed.

File: cogni-libs/src/main/java/zone/cogni/libs/sparqlservice/impl/FusekiConfig.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
+package zone.cogni.libs.sparqlservice.impl;
+
+import org.apache.commons.lang3.StringUtils;
+
+public class FusekiConfig extends Config {
+
+  private String updateUrl;
+  private String queryUrl;
+  private String graphStoreUrl;
```
Comment (natan-cox-cognizone) [inline cogni-libs/src/main/java/zone/cogni/libs/sparqlservice/impl/FusekiConfig.java:9]: are those 3 urls so different and unpredictable? can they not be based on the same "root" url?

File: cogni-libs/src/main/java/zone/cogni/libs/sparqlservice/impl/FusekiSparqlService.java [hunk 3]
```diff
@@ -76,35 +107,55 @@ public void executeUpdateQuery(String updateQuery) {
 
   @Override
   public void upload(Model model, String graphUri) {
```
Comment (alexey-lukashov) [inline cogni-libs/src/main/java/zone/cogni/libs/sparqlservice/impl/FusekiSparqlService.java:111]: is this upload replacing graph or updating with new values? one method we made in virtuoso became very handy with replace graph on demand parameter `public void upload(Model model, String graphUri, boolean replace)`

**Review by alexey-lukashov — APPROVED** [review]

File: cogni-libs/src/main/java/zone/cogni/libs/sparqlservice/impl/FusekiSparqlService.java (hunk 1 above)
Comment (domivds) [reply cogni-libs/src/main/java/zone/cogni/libs/sparqlservice/impl/FusekiSparqlService.java:34]: true
I propose we add that add as a future improvement. At same time when we do some clean up and deprecate one of the 2 sparql service versions.
Note, the advantage of the Provider is that we pass the type in the config. So the bean creation uses the provider and doesn't use a specific SparqlService class.

In mopo we have this as usage:
```  @Bean
  @MopoDataSource(MopoDataSource.Source.rdfstoreCuriaInput)
  public SparqlService sparqlServiceCuriaInput() {
    return sparqlServiceProvider().createSparqlService(MopoDataSource.Source.rdfstoreCuriaInput);
  }
```
  
  MopoDataSource.Source enum has 2 values (we use 2 different fuseki stores).

File: cogni-libs/src/main/java/zone/cogni/libs/sparqlservice/impl/FusekiConfig.java (hunk 2 above)
Comment (domivds) [reply cogni-libs/src/main/java/zone/cogni/libs/sparqlservice/impl/FusekiConfig.java:9]: In fuseki you can configure your different services with any URL you want. And this happened in mopo...

By default you can still just pass the main endpoint URL and the config will calculate the default values.

File: cogni-libs/src/main/java/zone/cogni/libs/sparqlservice/impl/FusekiSparqlService.java (hunk 3 above)
Comment (domivds) [reply cogni-libs/src/main/java/zone/cogni/libs/sparqlservice/impl/FusekiSparqlService.java:111]: good point
pushed an update for this

File: cogni-libs/src/main/java/zone/cogni/libs/sparqlservice/impl/FusekiSparqlService.java (hunk 1 above)
Comment (natan-cox-cognizone) [reply cogni-libs/src/main/java/zone/cogni/libs/sparqlservice/impl/FusekiSparqlService.java:34]: Cleanup of 2 sparql services and clean up of for services are configured can be handled separately.

File: cogni-libs/src/main/java/zone/cogni/libs/sparqlservice/impl/FusekiConfig.java (hunk 2 above)
Comment (natan-cox-cognizone) [reply cogni-libs/src/main/java/zone/cogni/libs/sparqlservice/impl/FusekiConfig.java:9]: weird. ok then 👍

**Review by natan-cox-cognizone — APPROVED** [review]

**Pull request merged**